### PR TITLE
feat(cli): generate changelog

### PR DESCRIPTION
**Pull request by robinjoseph08**
### What

Implements a changelog CLI that has the features that we expect it to have out-of-the-box.

### Why

So we can get off that old version of `conventional-changelog` and remove all the unnecessary scripts we have in multiple repos.

### Notes

* I think this is fully complete. You can try it out by cloning it, checking out this branch, and then running `../generate-changelog/bin/generate -h` in another repo. It makes a changelog that is very similar to ours but now has several new types:
  * `chore`
  * `docs`
  * `feat`
  * `fix`
  * `other`
  * `refactor`
  * `style`
  * `test`
* At first, I had it printing to stdout so that we can just `>>` into the `CHANGELOG.md` file, but I forgot that we (and most people) want it to prepend to the file instead of append, and since there's [no easy way to prepend](http://stackoverflow.com/a/15424018/2581607), I decided to build that functionality in. I left these as separate commits though mostly so that when `v1.0.0` is cut, the changelog generated for this repo will be more than one commit.
* I setup travis and coveralls from the get-go so this package started off fresh with 100% coverage.

Happy to hear any comments about style or functionality! I'm also down to do a peer code review as well.

@elnaz

### CODE REVIEW

File: test/writer.test.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+      return Writer.markdown(VERSION, commits, {})
+      .then(function (changelog) {
+        return changelog.split('\n');
+      })
+      .filter(function (line) {
+        return line.indexOf('#####') > -1;
+      })
+      .tap(function (lines) {
+        Expect(lines).to.have.length(1);
```
Comment (elnaz) [inline test/writer.test.js]: should there be two lines? one for each commit?

**Comment by elnaz** [comment]
This is awesome! What would be an example of a chore? I can't think of one that wouldn't be covered by docs, refactor, style, or test.

File: test/writer.test.js (hunk 1 above)
Comment (robinjoseph08) [inline test/writer.test.js]: This isn't looking for commits, it's looking for headings e.g. "New Features", "Bug Fixes", etc. This test is making sure that all `others` is grouped under the same heading of "Other Changes" as opposed to creating two headings, both of which say "Other Changes". Does that make sense?

**Comment by robinjoseph08** [comment]
@elnaz hmm I can't really think of anything off the top of my head, but these were the categories in [git-changelog](https://www.npmjs.com/package/git-changelog) so I thought it'd be a good starting point. It's there if we need it, but if we don't, then we won't use it.

**Comment by brianseitel** [comment]
Wanna update this for Node v5.1.1, too?

**Comment by brianseitel** [comment]
I might consider one-off data migrations (like backdating) or converting from controller -> plugin to be a chore... :)

**Comment by robinjoseph08** [comment]
@brianseitel I thought about it, but since it's open source, I'd rather it compatible with multiple versions of node. That's why I'm testing it again `0.10`, `0.12`, `4`, and `5` in travis.

**Comment by brianseitel** [comment]
:+1:

**Comment by brianseitel** [comment]
LGTM

**Comment by elnaz** [comment]
sounds good. LGTM

**Pull request merged**